libminifi/include/core/ContentRepository.h

```diff
@@ -53,10 +54,18 @@ class ContentRepository : public StreamManager<minifi::ResourceClaim>, public ut
 
   virtual void clearOrphans() = 0;
 
+  bool remove(const minifi::ResourceClaim &streamId) override {
+    return removeKey(streamId.getContentFullPath());
+  }
+
  protected:
+  void removeFromPurgeList();
+  virtual bool removeKey(const std::string& content_path) = 0;
+
   std::string directory_;
   std::mutex count_map_mutex_;
```

Comment:
rename this mutex as it no longer protects just the `count_map_` but the `purge_list_` as well

Reply:
I prefer having a separate mutex for the purge_list and separating the `count_map_` and  `purge_list_` processing blocks, updated in bde565c405e3e9419020445a48d41da561b58729